src/rvsdg/tree_unique/to_tree.rs

```diff
@@ -7,3 +7,245 @@
 //! These shared nodes need to be let-bound so that they are only
 //! computed once in the tree encoded
 //! program.
+#[cfg(test)]
+use crate::{cfg::program_to_cfg, rvsdg::cfg_to_rvsdg, util::parse_from_string};
+#[cfg(test)]
+use tree_unique_args::ast::{program, sequence};
+
+use crate::rvsdg::{BasicExpr, Id, Operand, RvsdgBody, RvsdgFunction, RvsdgProgram};
+use bril_rs::{Literal, ValueOps};
+use hashbrown::HashMap;
+use tree_unique_args::{
+    ast::{
+        add, arg, concat, function, get, num, print, program_vec, sequence_vec, tfalse, tlet, ttrue,
+    },
+    Expr,
+};
+
+impl RvsdgProgram {
+    pub fn to_tree_encoding(&self) -> Expr {
+        program_vec(
+            self.functions
+                .iter()
+                .map(|f| f.to_tree_encoding())
+                .collect(),
+        )
+    }
+}
+
+struct RegionTranslator<'a> {
+    /// The number of arguments to this region
+    num_args: usize,
+    /// a stack of let bindings to generate
+    bindings: Vec<Expr>,
+    /// A map from the rvsdg node id
+    /// to the index in the argument
+    /// where the value is stored.
+    /// After evaluating a node, do not evaluate it again.
+    /// Instead find it's index here.
+    index_of: HashMap<Id, usize>,
+    nodes: &'a Vec<RvsdgBody>,
+}
+
+fn cbind(expr: Expr, body: Expr) -> Expr {
```

Comment:
Comment?
Makes a new let binding, concat is just a convenience to avoid lots of nesting, but doesn't change semantics (right?)

Reply:
Yep, added a comment